src/Paket.Core/DependencyChangeDetection.fs

```diff
@@ -60,21 +58,113 @@ let findChangesInDependenciesFile(dependenciesFile:DependenciesFile,lockFile:Loc
     |> Seq.concat
     |> Set.ofSeq
 
-let GetUnchangedDependenciesPins (oldLockFile:LockFile) (changedDependencies:Set<GroupName*PackageName>) =
-    oldLockFile.GetGroupedResolution()
-    |> Seq.filter (fun kv -> not <| changedDependencies.Contains(kv.Key))
-    |> Seq.map (fun kv -> kv.Key, kv.Value.Version)
-    |> Map.ofSeq
+[<CustomEquality;CustomComparison>]
+type RemoteFileChange =
+    { Owner : string
+      Project : string
+      Name : string
+      Origin : ModuleResolver.SingleSourceFileOrigin
+      Commit : string option
+      AuthKey : string option }
+
+    override this.Equals(that) = 
+        match that with
+        | :? RemoteFileChange as that -> 
+            this.Owner = that.Owner &&
+             this.Project = that.Project &&
+             this.Name = that.Name &&
+             this.Origin = that.Origin &&
+             ((this.Commit = that.Commit) || this.Commit = None || that.Commit = None) &&
```

Comment:
that's the only interesting line here. the rest of CustomEquality and CustomComparison could be removed, but how?